shtc1/shtc1.c

```diff
@@ -46,50 +48,120 @@
 /* all measurement commands return T (CRC) RH (CRC) */
 #if USE_SENSIRION_CLOCK_STRETCHING
 static const u8 CMD_MEASURE_HPM[]     = { 0x7C, 0xA2 };
+static const u8 CMD_MEASURE_LPM[]     = { 0x64, 0x58 };
 #else
 static const u8 CMD_MEASURE_HPM[]     = { 0x78, 0x66 };
+static const u8 CMD_MEASURE_LPM[]     = { 0x60, 0x9C };
 static const u16 MEASUREMENT_DURATION_USEC = 14400;
 #endif /* USE_SENSIRION_CLOCK_STRETCHING */
 static const u8 CMD_READ_ID_REG[]     = { 0xef, 0xc8 };
 static const u8 COMMAND_SIZE = sizeof(CMD_MEASURE_HPM);
+
+static const u8 SHTC3_CMD_SLEEP[]     = { 0xB0, 0x98 };
+static const u8 SHTC3_CMD_WAKEUP[]    = { 0x35, 0x17 };
 #ifdef SHT_ADDRESS
 static const u8 SHTC1_ADDRESS = SHT_ADDRESS;
 #else
 static const u8 SHTC1_ADDRESS = 0x70;
 #endif
 
-static const u8 ID_REG_CONTENT    = 0x07;
-static const u8 ID_REG_MASK       = 0x1f;
+static const u16 SHTC1_PRODUCT_CODE_MASK    = 0x001F;
+static const u16 SHTC1_PRODUCT_CODE         = 0x0007;
+static const u16 SHTC3_PRODUCT_CODE_MASK    = 0x083F;
+static const u16 SHTC3_PRODUCT_CODE         = 0x0807;
+
+static const u8 *cmd_measure = CMD_MEASURE_HPM;
+
+/**
+ * PM_SLEEP is equivalent to
+ * if (ret) {
+ *     (void)sht_sleep(); // attempting to sleep but ignore return value
+ * } else {
+ *     (ret) = sht_sleep();
+ * }, (ret)
+ */
+#define PM_SLEEP(ret) ((ret) ? (sht_sleep(), (ret)) : ((ret) = sht_sleep()))
+
+/**
+ * PM_WAKE is equivalent to
+ * (ret) = sht_wakeup();
+ * if (ret)
+ *     (void)sht_sleep(); // attempting to sleep but ignore return value
+ * else {
+ *     (ret) = (cmd);
+ *     if (!(ret))
+ *          sht_sleep(); // attempting to sleep but ignore return value
+ * }, (ret)
+ */
+#define PM_WAKE(ret, cmd) (         \
```

Comment:
`cmd` is a little misleading, since its the return value of a command and not the command itself.

Reply:
Since it's a macro expansion, the command hasn't actually been executed yet.